support/Python/Machines.py

```diff
@@ -60,6 +65,22 @@ class Machine(yaml.YAMLObject):
     DefaultProcsPerNode: int
     DefaultQueue: str
     DefaultTimeLimit: str
+    LaunchCommandSingleNode: List[str]
+
+    def on_compute_node(self) -> bool:
+        """Determines whether or not we are running on a compute node."""
+        return os.environ.get("SLURM_JOB_ID") is not None
```

Comment:
I like this! Clever! :D And should be easy to update if we have to support anything other than slurm